### PR TITLE
Added the version info

### PR DESCRIPTION
**Pull request by alaGarbey675**
- Made the "in progress" status color lighter.
- Added the version of the application

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -4,7 +4,6 @@ import (
 	"github.com/datarootsio/cheek/cmd"
 )
 
-//go:generate npm run build
```
Comment (bart6114) [inline main.go:7]: Why don't we do this build step anymore? This was done so we have a fresh build of tailwind at every binary build.

File: pkg/http.go [hunk 2]
```diff
@@ -23,6 +23,12 @@ type Response struct {
 	Type   string `json:"type,omitempty"`
 }
 
+var version string // This will be injected at build time
```
Comment (bart6114) [inline pkg/http.go:26]: let's also set a `var commitSHA string` so we can easily link it back to a commit

File: pkg/http.go [hunk 3]
```diff
@@ -23,6 +23,12 @@ type Response struct {
...
+
+type VersionResponse struct {
+	Version string `json:"version"`
+}
```
Comment (bart6114) [inline pkg/http.go:30]: add the commit sha

File: pkg/web_assets/static/script.js [hunk 4]
```diff
@@ -137,4 +137,23 @@ function truncateDateTime(dateTimeStr) {
...
+    document.getElementById('version').textContent = `Version: ${data.version}`;
+  } catch (error) {
+    console.error('Fetch error:', error);
+  }
+}
+
+// Fetch version when the DOM is fully loaded
+document.addEventListener('DOMContentLoaded', () => {
+  fetchVersion(); // Call the function when the DOM is fully loaded
+});
```
Comment (bart6114) [inline pkg/web_assets/static/script.js:159]: no need to get the sha here

File: pkg/web_assets/static/styles.css [hunk 5]
```diff
@@ -3,4 +3,13 @@
   src: url('/static/GeistMonoVariableVF.woff2') format('woff2');
   font-weight: 100 900;
   font-style: normal;
+}
+#version {
```
Comment (bart6114) [inline pkg/web_assets/static/styles.css:7]: we should use tailwind to style this, not custom css

File: pkg/web_assets/templates/base.html [hunk 6]
```diff
@@ -1,51 +1,39 @@
...
+    <script src="//unpkg.com/alpinejs" defer></script>
+    <script type="text/javascript" src="/static/script.js"></script>
+  </head>
+  <body class="bg-slate-800">
+    <div class="max-w-4xl mx-auto">
+      <div class="flex pt-6 items-end">
+        <a class="drop-shadow-md text-3xl font-extrabold text-center bg-gradient-to-r from-lime-200 to-blue-400 bg-clip-text text-transparent hover:from-red-400 hover:via-yellow-300 hover:via-green-500 hover:to-lime-200 hover:bg-gradient-to-l hover:animate-pulse" href="/">cheek</a>
+        <div class="grow"></div>
+        <div class="col is-vertical-align is-right">
+          <div id="version" class="text-gray-400 text-sm">Version 1.0.0</div>
```
Comment (bart6114) [inline pkg/web_assets/templates/base.html:19]: let's add the version where it says 
image

So we get something like `shows statuses up until the last 10 runs (running cheek v1.0.0)` (just inject the version in a `span` dedicated to it).

**Review by bart6114 — COMMENTED** [review]
added some comments

can we also

- [ ] make these to be size `text-xs` (tailwind) https://github.com/datarootsio/cheek/blob/abdddf79384cabea351853067336dd9a7cc2dabb/pkg/web_assets/templates/base.html#L38 
- [ ] add the ci logic to set the build flags for the version/sha at buildtime

Thanks!

File: main.go (hunk 1 above)
Comment (alaGarbey675) [reply main.go:7]: That was removed by mistake. I was deleting a few lines of code that I had written, I will put it back. No changes were made to the main.go.

File: pkg/http.go (hunk 2 above)
Comment (alaGarbey675) [reply pkg/http.go:26]: // This will be injected at build time
var (
	version   string
	commitSHA string
)

File: pkg/http.go (hunk 3 above)
Comment (alaGarbey675) [reply pkg/http.go:30]: type VersionResponse struct {
    Version   string `json:"version"`
    CommitSHA string `json:"commit_sha"`
}

File: pkg/web_assets/static/styles.css (hunk 5 above)
Comment (alaGarbey675) [reply pkg/web_assets/static/styles.css:7]: "mt-4 p-2 bg-slate-800 text-slate-300 text-center rounded-md text-sm"

File: pkg/web_assets/templates/base.html (hunk 6 above)
Comment (alaGarbey675) [reply pkg/web_assets/templates/base.html:19]: Screenshot 2024-10-08 at 15 57 52

**Comment by alaGarbey675** [comment]
**build flags** : go build -ldflags="-X 'github.com/datarootsio/cheek/pkg.Version=${{ needs.version_tag.outputs.new_tag }}' -X 'github.com/datarootsio/cheek/pkg.CommitSHA=${{ steps.vars.outputs.sha_short }}'"

**Comment by bart6114** [comment]
Thank you @alaGarbey675 🙌

**Pull request merged**